src/import/sst_service.rs

```diff
@@ -675,6 +684,53 @@ impl<E: Engine> ImportSstService<E> {
     }
 }
 
+fn check_local_region_stale(
+    region_id: u64,
+    epoch: &RegionEpoch,
+    local_region_info: Option<RegionInfo>,
+) -> Result<()> {
+    match local_region_info {
+        Some(local_region_info) => {
+            let local_region_epoch = local_region_info.region.region_epoch.unwrap();
+
+            // TODO(lance6717): we should only need to check conf_ver because we require all
+            // peers have SST on the disk, and does not care about which one is
+            // leader. But since check_sst_for_ingestion also checks epoch version,
+            // we just keep it here for now.
+
+            // when local region epoch is stale, client can retry write later
+            if is_epoch_stale(&local_region_epoch, epoch) {
+                return Err(Error::Engine(
```

Comment:
Should the error type be `Engine`?

Reply:
https://github.com/tikv/tikv/pull/15795/commits/7c55600d6dcfd5079afb7fcbca6ed723c098828d